src/player.cpp

```diff
@@ -1732,8 +1759,15 @@ int player::run_cost(int base_cost, bool diag) const
     return int(movecost);
 }
 
+
 int player::swim_speed() const
 {
+    return swim_speed_encumbrance();
```

Comment:
Same as with move cost - `swim_speed_encumbrance` shouldn't include the base cost code, only the encumbrance-related one.